src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -638,8 +656,9 @@ static void close_transport_locked(grpc_exec_ctx *exec_ctx,
     while (grpc_chttp2_list_pop_writable_stream(t, &s)) {
       GRPC_CHTTP2_STREAM_UNREF(exec_ctx, s, "chttp2_writing:close");
     }
-    end_all_the_calls(exec_ctx, t, GRPC_ERROR_REF(error));
-    cancel_pings(exec_ctx, t, GRPC_ERROR_REF(error));
+    if (t->write_state == GRPC_CHTTP2_WRITE_STATE_IDLE) {
```

Comment:
Won't this be unconditionally true from  `if (t->write_state != GRPC_CHTTP2_WRITE_STATE_IDLE) {` on original line 605

Reply:
Right... that cleans things up a little.